### PR TITLE
Ability to search in DB logs.

### PR DESCRIPTION
**Pull request by seostudio**
Added search box in DB logs.

### CODE REVIEW

File: includes/admin/class-wc-admin-log-table-list.php [hunk 1]
```diff
@@ -334,6 +334,10 @@ protected function get_items_query_where() {
 			$where_conditions[] = 'source = %s';
 			$where_values[]     = wc_clean( $_REQUEST['source'] );
 		}
+		if ( ! empty( $_REQUEST['s'] ) ) {
+			$where_conditions[] = 'message like %s';
+			$where_values[]     = '%' . wc_clean( $_REQUEST['s'] ) . '%';
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-log-table-list.php]: This is unsafe.

Need to change this like to:

```php
$where_values[]     = '%' . $wpdb->esc_like( wc_clean( wp_unslash( $_REQUEST['s'] ) ) ) . '%';
```

File: includes/admin/views/html-admin-page-status-logs-db.php [hunk 2]
```diff
@@ -6,8 +6,17 @@
 if ( ! defined( 'ABSPATH' ) ) {
 	exit;
 }
+$s = '';
+if ( isset( $_REQUEST['s'] ) ) {
+	$s = wc_clean( $_REQUEST['s'] );
```
Comment (claudiosanches) [inline includes/admin/views/html-admin-page-status-logs-db.php]: Missing `wp_unslash()` before sanitization.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Requires a few fixes.

File: includes/admin/views/html-admin-page-status-logs-db.php (hunk 2 above)
Comment (seostudio) [reply includes/admin/views/html-admin-page-status-logs-db.php]: Done.

File: includes/admin/class-wc-admin-log-table-list.php [hunk 3]
```diff
@@ -334,6 +334,11 @@ protected function get_items_query_where() {
 			$where_conditions[] = 'source = %s';
 			$where_values[]     = wc_clean( $_REQUEST['source'] );
 		}
+		if ( ! empty( $_REQUEST['s'] ) ) {
+			$where_conditions[] = 'message like %s';
+			$s = wp_unslash( trim( $_REQUEST[ 's' ] ) );
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-log-table-list.php]: `wp_unslash()` needs to always comes first.

```
wp> wp_unslash( trim( '\\\\\\\Testing\\\\\\\\' ) )
=> string(11) "\\Testing\\"
wp> wp_unslash( trim( '\\         \\\\\Testing\\\\\       \\' ) )
=> string(11) " \Testing\ "
```

Just copy my example from before that will be just fine.

File: includes/admin/views/html-admin-page-status-logs-db.php (hunk 2 above)
Comment (claudiosanches) [reply includes/admin/views/html-admin-page-status-logs-db.php]: It's not done, still missing `wp_unslash()` here.

File: includes/admin/class-wc-admin-log-table-list.php (hunk 3 above)
Comment (claudiosanches) [reply includes/admin/class-wc-admin-log-table-list.php]: This still should be better:

```php
$where_values[]     = '%' . $wpdb->esc_like( wc_clean( wp_unslash( $_REQUEST['s'] ) ) ) . '%';
```

File: includes/admin/views/html-admin-page-status-logs-db.php (hunk 2 above)
Comment (seostudio) [reply includes/admin/views/html-admin-page-status-logs-db.php]: Now should be done :)

Comment (mikejolley) [reply includes/admin/views/html-admin-page-status-logs-db.php]: @seostudio it's still not done :p Look at the diff above.

Comment (seostudio) [reply includes/admin/views/html-admin-page-status-logs-db.php]: @mikejolley I apologize for the misunderstanding. Now is OK?

**Comment by claudiosanches** [comment]
Closing in favor of #17654
Thanks for your contribution.

**Pull request closed without merging**